### PR TITLE
add validations, move generate key functionality to developer_service

### PR DESCRIPTION
**Pull request by liambarstad**
#### What does  this PR do?
refactors the key generation so that it happens automatically as part of the DeveloperService. aforementioned object now has the .generate_key method open to further use
#### Questions:
  - Do Migrations Need to be ran? -> Yes
  - Do Environment Variables need to be set?
  - Any other deploy steps?

### CODE REVIEW

File: app/services/developer_service.rb [hunk 1]
```diff
@@ -6,9 +6,21 @@ def initialize(params)
 
   def register_as_developer
     if @user && !@user.developer?
-      ApiKey.create(user: @user)
-      @user.roles << Role.find_or_create_by(name: 'developer')
+      key = generate_key
```
Comment (liambarstad) [inline app/services/developer_service.rb]: @jmejia now raises an exception if the validation fails for whatever reason that can be caught elsewhere. I just put controller functionality within a begin/rescue block

Comment (jmejia) [reply app/services/developer_service.rb]: Much better. Nice 👍

File: app/services/developer_service.rb [hunk 2]
```diff
@@ -6,9 +6,21 @@ def initialize(params)
...
   end
 
+  def generate_key
+    Digest::SHA512.hexdigest(Time.now.to_s + @user.email)
+  end
+
+  private
+
+    def make_developer_role(user)
+      user.roles << Role.find_or_create_by(name: 'developer')
```
Comment (jmejia) [inline app/services/developer_service.rb]: @liambarstad Do we need to take in the user parameter on this method? Looks like we could use the user sent in in initialize.

Comment (jmejia) [reply app/services/developer_service.rb]: Speaking of... the parameter on initialize is called `params` but you only care about the user right now. I think we should change the parameter to `user`.

**Review by jmejia — COMMENTED** [review]
A few small changes but overall this looks pretty good.

**Pull request merged**